client/client.go

```diff
@@ -335,6 +343,15 @@ func (c *Client) processChannelRequests() {
 	c.l.Debug("Channel closed.")
 }
 
+func (c *Client) getActionTimeout(req *agentpb.StartActionRequest) *duration.Duration {
+	timeout := req.GetTimeout()
+	if reflect.DeepEqual(timeout, ptypes.DurationProto(0*time.Second)) || timeout == nil {
```

Comment:
Uh, no. Why we need reflect there? Unless there is something very tricky, it could be just:
```go
timeout := req.Timeout
if timeout == nil || timeout == ptypes.DurationProto(0)
```

Reply:
`ptypes.DurationProto` returns a pointer of type `duration.Duration` thats why